### DISCUSSION TITLE
PHP 8.2

### DISCUSSION THREAD

**Discussion by oortasertecfarma** [Jeroen-G/Explorer#150] [General]
When is it expected that Explorer can be used with PHP 8.2?

Thanks!

**Comment by Jeroen-G** [comment]
With your help, quite soon! I think these things are necessary:
- move 8.2 in CI from experimental to normal strategy
- add composer dependency
- test with an actual PHP8.2 app (I think I will upgrade the demo app too)

**Comment by oortasertecfarma** [comment]
I have updated composer but when running tests I get deprecation messages when trying to create dynamic properties in Mockery.

E.g.:
Deprecated: Creation of dynamic property Mockery_1_Laravel_Scout_Builder::$must is deprecated in /var/www/html/tests/Unit/ScoutSearchCommandBuilderTest.php on line 81

There is an open issue about this topic.
https://github.com/mockery/mockery/issues/1197

Do you have any ideas on how to adapt the tests?

**Reply by Jeroen-G** [reply]
Oh boy, that'll be fun 😅

Are the deprecation messages causing the tests to fail? If not, then fixing it can wait.

My first thoughts if they fail is that I have to research alternatives within mockery or alternatives to mockery.

So to answer: no I have no clear idea yet. If it is only a handful of tests and you don't know either then feel free to use `$this->markTestIncomplete('This test needs its mocks to be fixed for PHP8.2');`

**Comment by oortasertecfarma** [comment]
Hi Jeroen, 
No, the tests don't fail, they just show the messages.

PR with changes:
https://github.com/Jeroen-G/Explorer/pull/152/files

I have tested the app with 8.2 and it works.
https://github.com/Jeroen-G/explorer-demo/pull/6/files

**Reply by Jeroen-G** [reply]
Cool!

**Comment by Jeroen-G** [comment]
You may now beta test it by requiring `dev-master` of this package!
